hal/src/stm32f2xx/interrupts_hal.c

```diff
@@ -184,16 +190,22 @@ void HAL_Interrupts_Attach(uint16_t pin, HAL_InterruptHandler handler, void* dat
   NVIC_InitStructure.NVIC_IRQChannelCmd = ENABLE;
   //update NVIC registers
   NVIC_Init(&NVIC_InitStructure);
+  return SYSTEM_ERROR_NONE;
 }
 
-void HAL_Interrupts_Detach(uint16_t pin)
+int HAL_Interrupts_Detach(uint16_t pin)
 {
-  HAL_Interrupts_Detach_Ext(pin, 0, NULL);
+  return HAL_Interrupts_Detach_Ext(pin, 0, NULL);
 }
 
 
-void HAL_Interrupts_Detach_Ext(uint16_t pin, uint8_t keepHandler, void* reserved)
+int HAL_Interrupts_Detach_Ext(uint16_t pin, uint8_t keepHandler, void* reserved)
 {
+#if PLATFORM_ID == PLATFORM_ELECTRON_PRODUCTION
+  /* safety check that prevents users from attaching an interrupt to D7
+   * which is shared with BATT_INT_PC13 for power management */
+  if (pin == D7) return SYSTEM_ERROR_NOT_ALLOWED;
```

Comment:
Nitpicking: formatting.
```suggestion
  if (pin == D7) {
      return SYSTEM_ERROR_NOT_ALLOWED;
  }
```

Reply:
Fixed this formatting.